sdk/go/common/util/mapper/mapper_decode.go

```diff
@@ -88,13 +88,21 @@ func (md *mapper) DecodeValue(obj map[string]interface{}, ty reflect.Type, key s
 	if v, has := obj[key]; has {
 		// The field exists; okay, try to map it to the right type.
 		vsrc := reflect.ValueOf(v)
+	newValue:
 		// Ensure the source is valid; this is false if the value reflects the zero value.
 		if vsrc.IsValid() {
 			vdstType := vdst.Type().Elem()
 
+			// If the source is a ptr, dereference it as necessary to get the underlying
+			// value.
+			if vsrc.Type().Kind() == reflect.Ptr && !vsrc.IsNil() {
+				vsrc = vsrc.Elem()
+				goto newValue
+			}
+
```

Comment:
would recommend using a `while` loop where the `newValue` label is to unwrap the pointer ahead of time for processing rather than using a `goto`?

I suggest this because lines 92-95 can have new lines added in the future to them that may not be aware of the looping behavior.

Reply:
Sure